src/components/Input/BaseInputLayout/index.tsx

```diff
@@ -54,29 +56,41 @@ const BaseInputLayout = React.forwardRef<View, IBaseInputLayoutProps>(
       height: 0,
     });
 
-    const [rightContent, setRightContent] = useState(sideContent);
+    const [right, setRightContent] = useState(rightContent);
+    const [left, setLeftContent] = useState(leftContent);
 
     const { theme } = useTheme();
 
     const styles = useStyles();
 
-    useEffect(() => {
-      if (React.isValidElement(sideContent) && sideContent?.type === Button) {
-        let borderColor = theme.primary.main;
-        if (error) {
-          borderColor = theme.error.dark;
-        } else if (disabled) {
-          borderColor = theme.grey.light;
+    const updateContent = useCallback(
+      (onUpdate: (content: React.ReactElement) => void, content?: React.ReactElement) => {
+        if (React.isValidElement(content) && content?.type === Button) {
+          let borderColor = theme.primary.main;
+          if (error) {
+            borderColor = theme.error.dark;
+          } else if (disabled) {
+            borderColor = theme.grey.light;
+          }
+
+          const updatedContent = React.cloneElement(content, {
+            style: { borderColor: borderColor },
+            error: Boolean(error),
+            disabled: Boolean(disabled),
+          });
+          onUpdate(updatedContent);
```

Comment:
I think internal states for `right` and `left` content aren't required. If there are some states we need to handle inside of the content, they can be added from the part where we we add the components, for example `<Input leftContent={<SomeIcon disabled={disabled} />

Reply:
Changed according to today's discussing